### PR TITLE
[LYN-4065] Expose GameLift server api to notify server process ready

### PR DESCRIPTION
**Pull request by onecent1101**
## Details
As title descriped, add server public interface so multiplayer developer can decide when to notify server process on GameLift instance is ready.

## Testing
Pass unit test coverage 92% without gamelift sdk wrapper
Pass manual local testing with GameLiftLocal

Signed-off-by: onecent1101 <liug@amazon.com>

### CODE REVIEW

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp [hunk 1]
```diff
@@ -51,6 +66,37 @@ namespace AWSGameLift
         }
     }
 
+    GameLiftServerProcessDesc AWSGameLiftServerManager::BuildGameLiftServerProcessDesc()
+    {
+        GameLiftServerProcessDesc serverProcessDesc;
+        AZ::IO::FileIOBase* fileIO = AZ::IO::FileIOBase::GetDirectInstance();
+        if (fileIO)
+        {
+            const char pathToLogFolder[] = "@log@/";
```
Comment (junbo75) [inline Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp:75]: I'm wondering whether it makes sense to allow users to customize the log file paths. Developers may have additional log files they would like to upload like the files under user\eventlogger which are generated by event logger.

Maybe the API can accept an optional list of log file paths to override the default value.

Comment (onecent1101) [reply Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp:75]: good idea, but it should be a separate task

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp [hunk 2]
```diff
@@ -51,6 +66,37 @@ namespace AWSGameLift
...
+            }
+        }
+        else
+        {
+            AZ_Error(AWSGameLiftServerManagerName, false, "Failed to get File IO.");
+        }
+
+        if (auto console = AZ::Interface<AZ::IConsole>::Get(); console != nullptr)
+        {
+            [[maybe_unused]] AZ::GetValueResult getCvarResult = console->GetCvarValue("sv_port", serverProcessDesc.m_port);
```
Comment (junbo75) [inline Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp:93]: Curious: Why is this variable "maybe_unused"?

**Review by junbo75 — APPROVED** [review]
LG

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Include/Request/IAWSGameLiftServerRequests.h [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+#include <AzCore/EBus/EBus.h>
+#include <AzCore/RTTI/BehaviorContext.h>
+#include <AzCore/std/string/string.h>
+#include <AzFramework/Session/ISessionRequests.h>
+
+namespace AWSGameLift
+{
+    //! IAWSGameLiftServerRequests
+    //! Server interfaces to expose Amazon GameLift Server SDK
+    class IAWSGameLiftServerRequests
```
Comment (lmbr-pip) [inline Gems/AWSGameLift/Code/AWSGameLiftServer/Include/Request/IAWSGameLiftServerRequests.h:20]: Code style "We don't recommend using 'C','T','I' or any other letter to prefix a class"

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp [hunk 4]
```diff
@@ -51,6 +66,37 @@ namespace AWSGameLift
...
 
+    GameLiftServerProcessDesc AWSGameLiftServerManager::BuildGameLiftServerProcessDesc()
+    {
+        GameLiftServerProcessDesc serverProcessDesc;
+        AZ::IO::FileIOBase* fileIO = AZ::IO::FileIOBase::GetDirectInstance();
+        if (fileIO)
+        {
+            const char pathToLogFolder[] = "@log@/";
+            char resolvedPath[AZ_MAX_PATH_LEN];
+            if (fileIO->ResolvePath(pathToLogFolder, resolvedPath, AZ_ARRAY_SIZE(resolvedPath)))
```
Comment (lmbr-pip) [inline Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp:77]: Minor: Why do we need to resolve the path? What happens if the logs have not been generated yet and so the path does not exist?

**Review by lmbr-pip — COMMENTED** [review]
LG, mostly.

You have coverage numbers for tests

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Include/Request/IAWSGameLiftServerRequests.h (hunk 3 above)
Comment (onecent1101) [reply Gems/AWSGameLift/Code/AWSGameLiftServer/Include/Request/IAWSGameLiftServerRequests.h:20]: I am following the networking code style, where do you see that we should not have "I" to prefix a class?

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp (hunk 4 above)
Comment (onecent1101) [reply Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp:77]: It is copied from original place without any change.

**Review by lmbr-pip — APPROVED** [review]
LG

File: Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp [hunk 5]
```diff
@@ -99,12 +145,12 @@ namespace AWSGameLift
         return AZ::IO::Path();
     }
 
-    bool AWSGameLiftServerManager::InitializeGameLiftServerSDK()
+    void AWSGameLiftServerManager::InitializeGameLiftServerSDK()
```
[the remark this answers is not in the document]
Comment (onecent1101) [reply Gems/AWSGameLift/Code/AWSGameLiftServer/Source/AWSGameLiftServerManager.cpp:148]: as it is an internal function not triggered by customer (the caller is gamelift system component), and the state of initialization is maintained internally with private field m_serverSDKInitialized.

**Pull request merged**